### ISSUE TITLE
Predicton step that can be run in between training epochs

### ISSUE THREAD

**Issue by OlfwayAdbayIgbay** [PyTorchLightning/lightning-flash#643]
## 🚀 Feature
Add an "inference" step, or a keyword to trainer.predict() which doesn't remove the model from GPU after the prediction run.

### Motivation

For things like online triplet mining, I am predicting batches of data in between epochs. I don't need gradients for those, so I use self.trainer.predict.

something curious I've noted, is that after running "trainer.predict", the model is removed from the cuda device. I assume that is wanted behaviour, so right now I fix that by running "self.cuda(gpu_index)" afterwards.

### Pitch

Since lightning is all about keeping code beautiful and organized, I propose there should be a keyword in trainer.predict() which keeps the model on the cuda device.

### Alternatives

Right now, it works to write "self.cuda(gpu_index)"  at the end of the prediction loop. However, if one doesn't know this, then it can take quite a while to figure out, especially in larger projects.

**Comment by ethanwharris** [comment]
Hi @OlfwayAdbayIgbay thanks for reporting this! Could you open an issue on the PL github for this? I think it's something that should be fixed over there. The fix would just be to put the model back to the device it started on rather than sending it to CPU.

**Comment by OlfwayAdbayIgbay** [comment]
Oh sorry, I don't know how I ended up in the wrong project!

**Comment by ethanwharris** [comment]
@OlfwayAdbayIgbay No worries :smiley: Closing this
